Understanding Confidence Intervals:

A confidence interval estimates the range in which a population parameter (like the mean) is expected to fall, given a sample statistic.

When the sample size is small and the population standard deviation is unknown, we use the t-distribution to estimate the confidence interval.

When the population standard deviation is known, we can use the z-distribution to estimate the confidence interval.

Scenario:

We are tasked with constructing two different confidence intervals for the mean durability of print-heads:

*Part a: Using the sample standard deviation and the t-distribution.

*Part b: Using the known population standard deviation and the z-distribution.

Step-by-Step Approach:

1.Import the necessary libraries.

2.Calculate the sample statistics (mean, sample standard deviation).

3.Use the t-distribution to compute the confidence interval in part (a)

4.Use the z-distribution to compute the confidence interval in part (b).

In [1]:
# Import necessary libraries
import numpy as np
import scipy.stats as stats


In [9]:
# Data: Durability of print-heads (in millions of characters)
data = np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48, 1.20, 1.33, 1.18, 1.22, 1.29])

In [3]:
# Step 1: Calculate the sample statistics
sample_mean = np.mean(data)  # Sample mean
sample_std = np.std(data, ddof=1)  # Sample standard deviation (using ddof=1 for sample)
n = len(data)  # Sample size

print(f"Sample Mean: {sample_mean}")
print(f"Sample Standard Deviation: {sample_std}")
print(f"Sample Size: {n}")

Sample Mean: 1.2386666666666666
Sample Standard Deviation: 0.19316412956959936
Sample Size: 15


In [6]:
# Part a: 99% Confidence Interval Using Sample Standard Deviation (t-distribution)
confidence_level = 0.99
alpha = 1 - confidence_level
df = n - 1  # Degrees of freedom for the t-distribution
t_critical = stats.t.ppf(1 - alpha/2, df)  # t-critical value for 99% confidence

# Margin of error for t-distribution
margin_of_error_t = t_critical * (sample_std / np.sqrt(n))

# Confidence interval using sample standard deviation
ci_lower_t = sample_mean - margin_of_error_t
ci_upper_t = sample_mean + margin_of_error_t

print("\n99% Confidence Interval using Sample Standard Deviation (t-distribution):")
print(f"Lower bound: {ci_lower_t}")
print(f"Upper bound: {ci_upper_t}")



99% Confidence Interval using Sample Standard Deviation (t-distribution):
Lower bound: 1.0901973384384906
Upper bound: 1.3871359948948425


In [7]:
# Part b: 99% Confidence Interval Using Known Population Standard Deviation (z-distribution)
pop_std = 0.2  # Known population standard deviation
z_critical = stats.norm.ppf(1 - alpha/2)  # z-critical value for 99% confidence


In [8]:
# Margin of error for z-distribution
margin_of_error_z = z_critical * (pop_std / np.sqrt(n))

# Confidence interval using known population standard deviation
ci_lower_z = sample_mean - margin_of_error_z
ci_upper_z = sample_mean + margin_of_error_z

print("\n99% Confidence Interval using Known Population Standard Deviation (z-distribution):")
print(f"Lower bound: {ci_lower_z}")
print(f"Upper bound: {ci_upper_z}")


99% Confidence Interval using Known Population Standard Deviation (z-distribution):
Lower bound: 1.1056514133957607
Upper bound: 1.3716819199375725


Explanation of the Code:

*numpy is used for basic numerical operations.

*scipy.stats is used to compute critical values for the t-distribution and z-distribution.

Sample Statistics:

We calculate the sample mean (np.mean(data)) and the sample standard deviation (np.std(data, ddof=1)), where ddof=1 ensures we use the sample formula.

Part a (Using the t-distribution):

Since the sample size is small and the population standard deviation is unknown, we use the t-distribution.

We calculate the degrees of freedom (df = n - 1) and the t-critical value using stats.t.ppf() for a 99% confidence level.

Part b (Using the z-distribution):

When the population standard deviation is known, we use the z-distribution.

We compute the z-critical value using stats.norm.ppf() for a 99% confidence level.

Output:

The code prints the 99% confidence interval for both cases (using the sample standard deviation and using the known population standard deviation).

Sample OutPut:

Sample Mean: 1.2386666666666666

Sample Standard Deviation: 0.19316412956959936

Sample Size: 15


99% Confidence Interval using Sample Standard Deviation (t-distribution:

Lower bound: 1.0901973384384906

Upper bound: 1.3871359948948425

99% Confidence Interval using Known Population Standard Deviation (z-distribution):

Lower bound: 1.1056514133957607

Upper bound: 1.3716819199375725